### PR TITLE
[Tizen] Vscode debug

### PR DESCRIPTION
**Pull request by jlatusek**
#### Problem
There is no support for debugging a Tizen app 

#### Change overview
* New tasks for vscode to build, launch gdbserver, install the app
* update Lighting App dependencies in Tizen container image
* fix symbolic link creation
* Add missing sdb tool

#### Testing
Use new tasks to build, install, launch gdbserver and debug a Tizen app.

Build tizen docker image:
```sh
./integrations/docker/images/chip-build-tizen/build.sh
```

### CODE REVIEW

File: integrations/docker/images/chip-build-tizen/Dockerfile [hunk 1]
```diff
@@ -44,6 +44,7 @@ RUN set -x \
     http://download.tizen.org/sdk/tizenstudio/official/binary/certificate-generator_0.1.3_ubuntu-64.zip \
     http://download.tizen.org/sdk/tizenstudio/official/binary/new-common-cli_2.5.7_ubuntu-64.zip \
     http://download.tizen.org/sdk/tizenstudio/official/binary/new-native-cli_2.5.7_ubuntu-64.zip \
+    http://download.tizen.org/sdk/tizenstudio/official/binary/sdb_4.2.23_ubuntu-64.zip \
```
Comment (andy31415) [inline integrations/docker/images/chip-build-tizen/Dockerfile:47]: docker file updates require a updated version file.

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+POSITIONAL_ARGS=()
+APP_ARGS=""
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --app-name)
+            APP_NAME="$2"
+            shift
+            shift
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:21]:
```suggestion
            shift 2
```
In other places as well

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+SDB="$TIZEN_SDK_ROOT/tools/sdb"
+PORT=9999
+RESULT_MODE="debug"
+
+function help() {
+    cat <<EOF
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:8]: Why not a good old `echo` ? Also, I would use "$0" without basename, unless this script is on the PATH.

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+SDB="$TIZEN_SDK_ROOT/tools/sdb"
+PORT=9999
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:4]: Can we make this port configurable? E.g.: debugging two apps at the same time.

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+SDB="$TIZEN_SDK_ROOT/tools/sdb"
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:3]: You've added `$TIZEN_SDK_ROOT/tools` to the PATH, so I think that simple `sdb` will be good enough.

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --app-name)
+            APP_NAME="$2"
+            shift
+            shift
+            ;;
+        --help)
+            help
+            shift
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:25]: I don't think that we should run this script after `--help`. 
```suggestion
            exit 0
```

**Review by arkq — CHANGES_REQUESTED** [review]

File: scripts/helpers/tizen_gdbserver_run.sh (hunk 3 above)
Comment (jlatusek) [reply scripts/helpers/tizen_gdbserver_run.sh:8]: I've forgotten to add a description to arguments, and I think that multiline help it look cleaner.

File: scripts/helpers/tizen_gdbserver_run.sh (hunk 4 above)
Comment (jlatusek) [reply scripts/helpers/tizen_gdbserver_run.sh]: Fixed, also I've added possibility do choose target device

File: scripts/helpers/tizen_gdbserver_run.sh (hunk 5 above)
Comment (jlatusek) [reply scripts/helpers/tizen_gdbserver_run.sh]: Fixed

File: .vscode/launch.json [hunk 7]
```diff
@@ -387,6 +387,39 @@
                 "/tmp/ota-image.bin"
             ],
             "cwd": "${workspaceFolder}"
+        },
+        {
+            "name": "LightingApp (Tizen)[remote]",
```
Comment (Arkq) [inline .vscode/launch.json:392]:
```suggestion
            "name": "LightingApp (Tizen) [remote]",
```
And maybe move this entry under the Linux one?

File: scripts/helpers/tizen_gdbserver_run.sh [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+if [ "$(sdb devices | wc -l)" -le 1 ]; then
+    echo "No device connected"
+    exit 1
+fi
+
+CMD="/usr/bin/app_launcher -s \"$APP_NAME\"  \
+__AUL_SDK__ $RESULT_MODE \
+__DLP_DEBUG_ARG__ :$GDBSERVER_PORT \
+__DLP_GDBSERVER_PATH__ $SDK_TOOLS_PATH/gdbserver/gdbserver \
```
Comment (Arkq) [inline scripts/helpers/tizen_gdbserver_run.sh:70]: From where is this `$SDK_TOOLS_PATH` variable?

**Review by arkq — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**